components/automate-gateway/api/nodes/nodes.proto

```diff
@@ -142,96 +265,146 @@ service NodesService {
 	bool ssl = 27;
+	// Allow self-signed certificate (boolean).
 	bool self_signed = 28;
+	// Username from the credential ID for this node.
 	string user = 29;
-	repeated string key_files = 31;
+	// Sudo options to use when accessing the node.
 	string sudo_options = 33;
-	string region = 34;
-	string subscription_id = 35;
+	// List of hostnames (FQDN or IP address) for bulk creating nodes.
 	repeated string hosts = 40;
 }
 
+// Node information.
 message Node {
+	// Unique node ID (UUID).
 	string id = 1;
+	// User-specified node name.
 	string name = 2;
+	// Node platform.
 	string platform = 3;
+	// Node platform version.
 	string platform_version = 4;
+	// Node manager (automate, aws-ec2, aws-api, azure-vm, azure-api, gcp).
 	string manager = 5;
+	// Node tags.
 	repeated chef.automate.domain.compliance.api.common.Kv tags = 20;
+	// Timestamp of the last `detect` or `exec` job.
 	google.protobuf.Timestamp last_contact = 21;
+	// Node status (unreachable, reachable, unknown).
 	string status = 22;
+	// Results of the last compliance scan job for this node.
 	ResultsRow last_job = 23;
+	// Node configuration for ssh or winrm.
 	TargetConfig target_config = 99;
+	// List of manager IDs for the node.
 	repeated string manager_ids = 24;
+	// Last connection error received when trying to contact the node.
 	string connection_error = 25;
+	// Last known node state (running, stopped, terminated).
 	string state = 26;
+	// Prefix for node name. The full node name is the prefix + the host.
 	string name_prefix = 27;
+	// List of projects associated with the node. Projects are a concept introduced in IAMv2.
 	repeated string projects = 28;
+	// Most recent node data from the last Chef Infra run results. 
 	LastContactData run_data = 29;
+	// Most recent compliance scan data for the node from the last InSpec scan. 
 	LastContactData scan_data = 30;
 }
 
+// Most recent node data from the latest Chef Infra run and InSpec scan.
 message LastContactData {
-	string id = 1; // run id or report id
+	// Chef Infra run report ID or InSpec scan report ID.
+	string id = 1;
 	enum Status {
 		UNKNOWN = 0;
 		PASSED = 1;
 		FAILED = 2;
 		SKIPPED = 3;
 	}
+	// Last node report status.
 	Status status = 2;
+	// Next-to-last node status report.
 	Status penultimate_status = 3;
+	// Last node report endtime.
 	google.protobuf.Timestamp end_time = 4;
 }
 
 message Nodes {
+	// List of nodes.
 	repeated Node nodes = 1;
+	// Total number of nodes in the system.
 	int32 total = 20;
+	// Total number of unreachable nodes in the system.
 	int32 total_unreachable = 21;
+	// Total number of reachable nodes in the system.
 	int32 total_reachable = 22;
+	// Total number of unknown nodes in the system.
 	int32 total_unknown = 23;
 }
 
+// Summary details from the most recent scan job that executed for the node.
 message ResultsRow {
+	// Node ID for the result.
 	string node_id = 1;
+	// Report ID for the result.
 	string report_id = 2;
+	// Status of the report.
```

Comment:
List the possibilities:

`Report status: successful, failed, error` or something like that